src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs

```diff
@@ -170,11 +163,19 @@ public async Task ExecuteAsync(ExportJobRecord exportJobRecord, WeakETag weakETa
                 // Try to update the job to failed state.
                 _logger.LogError(ex, "Encountered an unhandled exception. The job will be marked as failed.");
 
-                await UpdateJobStatus(OperationStatus.Failed, updateEndTimestamp: true, cancellationToken);
+                await UpdateAndCommitJobStatus(OperationStatus.Failed, updateEndTimestamp: true, cancellationToken);
+            }
+        }
+
+        private void UpdateJobOutputData(IDictionary<string, ExportFileInfo> resourceMapping)
+        {
+            foreach (KeyValuePair<string, ExportFileInfo> kvp in resourceMapping)
+            {
+                _exportJobRecord.Output.TryAdd(kvp.Key, kvp.Value);
```

Comment:
Couldn't this be added after line 234? Since the only time you will add a new entry to Output is when you create a new file?